codes/depression_among_adults.qmd

```diff
@@ -7,119 +7,114 @@ date: "2025-02-15"
 
 ## Project Description
 
-Provide the project description here.
+This dataset contains information on adult depression among males and females from different ethnicities in the USA.
+
+## Research questions
+
+- Which ethnic group has the highest prevalence of depression over the years?
+
+- Is there a significant difference in depression rates based on income levels
+Statistical Association
+
+- Is there a correlation between weighted frequency and percentage of depression?
+
+- Are there significant differences in depression rates between different age groups ?
+
+- Show the distribution of depression rates across different groups.
+
 
 ## Loading of Libraries 
 
-Provide instructions for library installation and reason why you are using every library.
+For the library installation I prefer to use the {pacman} package . It is a package manager that manages and facilitate the installation of other packages. 
+
 
 ```{r setup, include=FALSE}
-library(readr)
-library(ggplot2)
-library(tidyr)
-library(dplyr)
-library(nortest)
-library(magrittr)
+if (!require(readr)) install.packages("readr") or 
+
+# if(!require(pacman)) install.packages("pacman")
+# pacman::pload(readr, ggplot2, tidyr,dplyr, nortest, magrittr)
+library(readr) # It is used for reading and writing data efficiently
+library(ggplot2) # Used for data visualization
+library(tidyr) # this pacakge is use for data cleaning and reshaping 
+library(dplyr) # They are used for data manipulation and transformation
+library(nortest) # Provides multiple statistical tests to check if a dataset follows a normal distribution
+library(magrittr) # used for writting cleaner, more readable code by providing the pipe operator.
 ```
 
 ## Import dataset
 
+The `dirname()` function in R extracts the directory (folder ) path from a given file path. It is useful when working with file paths, directories. 
+
 ```{r}
 #Set working directory
 
-setwd("/Users/admin/Desktop/adult-depression-lghc-indicator-24.csv")
-getwd()
-# create a file_path for easy access
+# the followings are not needed - commented them out
+# setwd("/Users/admin/Desktop/adult-depression-lghc-indicator-24.csv")
+# getwd()
 
-file_path <- file.path(getwd(), "adult-depression-lghc-indicator-24.csv")
+# create a file_path for easy access
+file_path <- file.path(dirname(getwd()), "depression_data", "adult-depression-lghc-indicator-24.csv")
 
+# import the data
 depression_data <- read.csv(file_path)
 ```
 
 ## Description of the dataset 
```

Comment:
Check the data source to know the proper definition of each column. Specially, you need to know exactly what you have in column 5 to 7